src/icp_canister_backend/tests/test.rs

```diff
@@ -1424,3 +1424,42 @@ fn test_reward_rate_increase_decrease_during_episodes() {
         decreased_reward_rate
     );
 }
+
+#[test]
+fn test_reward_distribution_middle_and_final() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
+    let deposit_amount = Nat::from(100_000_000u64);
+    let reward_amount = Nat::from(50_000_000u64);
+
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 0);
+    create_deposit(&pic, canister_id, ledger_id, user, deposit_amount.clone(), stakable_episode);
+    reward_pool(&pic, canister_id, ledger_id, user, reward_amount.clone())
+        .expect("Reward pool should succeed");
+
+    let earning_duration = get_episode_time_to_end(&pic, stakable_episode);
+
+    //Middle  rewards distribution
+    advance_time(&pic, earning_duration / 2);
+    
+    let middle_rewards = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get middle rewards");
+    
+    let expected_middle = Nat::from(earning_duration / 2);
+    let allowed_error_middle = expected_middle.clone() / Nat::from(100u64);
+    assert_with_error(&middle_rewards, &expected_middle, &allowed_error_middle, "Middle reward distribution");
+
+    //  End  rewards distributed
+    advance_time(&pic, earning_duration / 2);
+    
+    let final_rewards = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get final rewards");
+    
+    let expected_final = Nat::from(earning_duration);
+    let allowed_error_final = expected_final.clone() / Nat::from(100u64);
+    assert_with_error(&final_rewards, &expected_final, &allowed_error_final, "Final reward distribution");
```

Comment:
I would expect the final distributed rewards to be equal all of the rewards.

Reply:
Also it would be probably better to change in reward pool in the line
let reward_duration = 365 * 24 * 60 * 60 + (current_episode_finish_time - current_time);
365 * 24 * 60 * 60 to EPISODE_DURATION * 12
It will make a better accounting for the reward tests.